### PR TITLE
Lua API formatting and grammar fixes

### PR DESCRIPTION
**Pull request by pauloue**
* ~When generating HTML the title gets included in the table of contents and it's not necessary anyway, so I removed the title section~
* Used tree output for the directory structure
* Spelling/grammar fixes
* Use headings and lists for mapgen aliases so they aren't all on one line when generated to HTML
* Fix skipped heading levels

### CODE REVIEW

**Comment by ClobberXD** [comment]
I feel that the title should still be present... Is there no other way to exclude the title from the TOC?

**Comment by SmallJoker** [comment]
`Introduction` could be replaced with `Minetest Lua Modding API Reference` to still have a title. It would quite imply the start of the introduction already.

**Comment by rubenwardy** [comment]
My modding book removes the header before making the table of contents

**Comment by paramat** [comment]
I agree with SmallJoker's idea. I was intending to remove the unnecessary space at file start anyway, so could you just remove the 4 empty lines and 'introductionm'? Then i'll +1.
The tree structure changes are good.

**Comment by paramat** [comment]
I disagree with removing the document title. I was actually intending to reduce the 4 empty lines at file start to 2. The tree structure changes are good.

File: doc/lua_api.txt [hunk 1]
```diff
@@ -921,9 +919,9 @@ are quite static.
 
 The definition of a node is stored and can be accessed by name in
```
Comment (paramat) [inline doc/lua_api.txt]: `The definition of a node is stored and can be accessed by using`

File: doc/lua_api.txt [hunk 2]
```diff
@@ -921,9 +919,9 @@ are quite static.
...
 
-    minetest.registered_nodes[node.name]
```
Comment (paramat) [inline doc/lua_api.txt]: .. then leave this as it is, it's useful to see how to get the nodedef.

File: doc/lua_api.txt [hunk 3]
```diff
@@ -2145,8 +2143,8 @@ Player Inventory lists
 
 
 
-Colours
-=======
+Colors
```
Comment (paramat) [inline doc/lua_api.txt]: Oops thanks, my mistake.

**Comment by paramat** [comment]
Apart from line comments all else looks good.

**Comment by paramat** [comment]
Yes i considered that too, but for a later PR after some discussion.
The reason i used `=====` under all section titles was to make them stand out more in .txt. form and differentiate them from `-----` titles.
I also used `=====` because in MD format it made the section title font as large as the file title, making sections easier to find when scrolling through and clearly differentiating them from subsections.
I think that readability in .txt form should not be reduced any further for the sake of strict MD format, there has to be more of a balance.
Maybe we can add `#` or '##` to section titles underlined with `=====`?

**Comment by paramat** [comment]
To avoid conflicts i'd like to get this merged tonight in it's present state, if that's ok with you? Large lua_api.txt PRs tend to suffer from conflicts arising.

**Comment by paramat** [comment]
Hmm if i do:
```
#Introduction
==========
```
The font becomes large but then the `=====` has to be removed as it is no longer hidden, which means in .txt form it would have to be:
```
#Introduction
```
Which is not as clear. This is probably why i edited the file as i did, even though strictly it should have levels of `#`.

**Comment by paramat** [comment]
Similar issue with:
```
#Startup
-------
```
Instead of the '-----' being hidden it becomes a line across the document which is unsuitable for a subsection.

**Comment by paramat** [comment]
Note: `####` results in a bold title, but `#####` results in a tiny font smaller than the normal font so can't be used, unfortunately it is used somewhere.

**Comment by pauloue** [comment]
No reason to use both hashes and underlines -- a `#` and underlining with `===` are two different ways of making an `<h1>`, and `##` and an underline with `---` are both `<h2>`s. I agree with always using `===` for `<h1>`s and `---` for `<h2>`s.

**Comment by pauloue** [comment]
Yes, in fact I think all the headings in below the Class Reference heading should be a level higher because it skips from an h1 to an h3.

**Comment by paramat** [comment]
So it seems the document is mostly like this and i suggest this should be the desired form:

h1 ===== : Document title and section titles.
h2 ----- : Subsection titles.
h3 ###
h4 ####
(no h5 #####)

**Comment by paramat** [comment]
> Yes, in fact I think all the headings in below the Class Reference heading should be a level higher

Fine for me.

**Comment by paramat** [comment]
That could be done for `'minetest' namespace reference` and `Definition tables` sections too.

**Comment by paramat** [comment]
Ok will review tonight and will probably +1, will try to merge it tonight to avoid conflicts tomorrow (EU time).

File: doc/lua_api.txt [hunk 4]
```diff
@@ -1535,19 +1553,22 @@ Tools define:
         * Digging times
         * Damage groups
 
-#### Full punch interval
+Full punch interval
```
Comment (paramat) [inline doc/lua_api.txt]: This is actually `###` a level below `Tools definition`as it describes a part of the tool definition.

File: doc/lua_api.txt [hunk 5]
```diff
@@ -1535,19 +1553,22 @@ Tools define:
...
+-------------------
 When used as a weapon, the tool will do full damage if this time is spent
 between punches. If e.g. half the time is spent, the tool will do half
 damage.
 
-#### Maximum drop level
+Maximum drop level
```
Comment (paramat) [inline doc/lua_api.txt]: This is actually `###` a level below `Tools definition`as it describes a part of the tool definition.

File: doc/lua_api.txt [hunk 6]
```diff
@@ -1535,19 +1553,22 @@ Tools define:
...
+------------------
 Suggests the maximum level of node, when dug with the tool, that will drop
 it's useful item. (e.g. iron ore to drop a lump of iron).
 
 This is not automated; it is the responsibility of the node definition
 to implement this.
 
-#### Uses
+Uses
```
Comment (paramat) [inline doc/lua_api.txt]: This is actually `###` a level below `Tools definition`as it describes a part of the tool definition.

File: doc/lua_api.txt [hunk 7]
```diff
@@ -1556,11 +1577,13 @@ is multiplied by `3^leveldiff`.
 * `uses=10, leveldiff=1`: actual uses: 30
 * `uses=10, leveldiff=2`: actual uses: 90
 
-#### Maximum level
+Maximum level
```
Comment (paramat) [inline doc/lua_api.txt]: This is actually `###` a level below `Tools definition`as it describes a part of the tool definition.

File: doc/lua_api.txt [hunk 8]
```diff
@@ -1556,11 +1577,13 @@ is multiplied by `3^leveldiff`.
...
+-------------
 Tells what is the maximum level of a node of this group that the tool will
 be able to dig.
 
-#### Digging times
+Digging times
```
Comment (paramat) [inline doc/lua_api.txt]: This is actually `###` a level below `Tools definition`as it describes a part of the tool definition.

File: doc/lua_api.txt [hunk 9]
```diff
@@ -1573,10 +1596,12 @@ If the result digging time is 0, a delay of 0.15 seconds is added between
 digging nodes; If the player releases LMB after digging, this delay is set to 0,
 i.e. players can more quickly click the nodes away instead of holding LMB.
 
-#### Damage groups
+Damage groups
```
Comment (paramat) [inline doc/lua_api.txt]: This is actually `###` a level below `Tools definition`as it describes a part of the tool definition.

File: doc/lua_api.txt [hunk 10]
```diff
@@ -3063,27 +3101,33 @@ callback, `nil` is returned.
 
 The following Mapgen objects are currently available:
 
-### `voxelmanip`
+`voxelmanip`
```
Comment (paramat) [inline doc/lua_api.txt]: This is a list of mapgen objects, not section titles, best leave these 6 unchanged or change to a list.

**Comment by paramat** [comment]
Apart from the above line comments (which could be fixed on merge if authot doesn't) +1.

**Comment by paramat** [comment]
:+1: WIll merge in an hour.

**Pull request merged**